core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala

```diff
@@ -532,6 +559,15 @@ private[netty] class NettyRpcEndpointRef(
     nettyEnv.ask(new RequestMessage(nettyEnv.address, this, message), timeout)
   }
 
+  override def askWithCancelCheck[T: ClassTag](
+      message: Any,
+      timeout: RpcTimeout,
+      isInterrupted: () => Boolean,
+      checkInterruptInterval: Long): Future[T] = {
+    nettyEnv.askWithInterruptCheck(new RequestMessage(nettyEnv.address, this, message),
```

Comment:
Nit: Also keeping the same name `askWithCancelCheck`?

Reply:
Let's discuss my updated code which make interface more simpler.